### DISCUSSION TITLE
UserWarning: The name 'vnc_bp' is already registered for this blueprint. Use 'name=' to provide a unique name. This will become an error in Flask 2.1.

### DISCUSSION THREAD

**Discussion by nmz787-intel** [pallets/flask#4669] [Q&A]
I have added a print statement inside `flask.blueprints.register` and I only see one printout for my "vnc_bp"... yet I still get this error. I've had to retain an older version that still just warns, rather than raises.

**Comment by nmz787-intel** [comment]
The blueprints are being registered through this websocket middleware:
https://github.com/heroku-python/flask-sockets/

**Comment by davidism** [comment]
I can't reproduce this issue with the information provided. I'd look at what `flask.blueprints` contains as you register your blueprints, to start debugging. Likely you've got two blueprints with the same name somewhere along the line.

**Reply by nmz787-intel** [reply]
`app.blueprints` doesn't change after I register the flask-socket blueprints.
